ssm/message_directory.py

```diff
@@ -0,0 +1,138 @@
+"""
+Copyright (C) 2018 STFC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+@author: Greg Corbett
+"""
+
+import logging
+import os
+import uuid
+
+# logging configuration
+LOG = logging.getLogger(__name__)
+
+
+class MessageDirectory(object):
+    """A structure for holding Accounting messages in a directory."""
+
+    def __init__(self, path):
+        """Create a new directory structure for holding Accounting messages."""
+        self.directory_path = path
+
+    def add(self, data):
```

Comment:
We don't actually use `add` for the outgoing queue. Could drop it?

Reply:
Well, we could have a stub class that raises a useful error, just in case.

Reply:
We don't use `dirq.add` at the moment. But I find the `dirq.add` functionality quite useful for running manual tests. And maybe in the future we could automate some of those tests and make use of this `add` method

Reply:
Fair enough. Was just thinking of avoiding complexity.